### PR TITLE
SEQNG-1104 Send N&S pause and stop command state to web client.

### PR DESCRIPTION
**Pull request by jluhrs**
The step information send to the client now indicates if there is a stop gracefully or pause gracefully command in progress.

### CODE REVIEW

File: modules/seqexec/model/shared/src/main/scala/seqexec/model/Step.scala [hunk 1]
```diff
@@ -212,7 +211,8 @@ final case class NodAndShuffleStep(
   override val skip:       Boolean,
   override val fileId:     Option[ImageFileId],
   configStatus:            List[(Resource, ActionStatus)],
-  nsStatus:                NodAndShuffleStatus
+  nsStatus:                NodAndShuffleStatus,
+  pendingObserveCmd:       Option[NodAndShuffleStep.PendingObserveCmd]
```
Comment (jluhrs) [inline modules/seqexec/model/shared/src/main/scala/seqexec/model/Step.scala]: Posible pending N&S observation command.

File: modules/seqexec/server/src/main/scala/seqexec/server/StepsView.scala [hunk 2]
```diff
@@ -101,7 +103,7 @@ object StepsView {
 
   private def observeAction[F[_]](executions: List[ParallelActions[F]]): Option[Action[F]] =
     // FIXME This is too naive and doesn't work properly for N&S
-    executions.flatMap(_.toList).filter(_.kind === ActionType.Observe).headOption
+    executions.flatMap(_.toList).find(_.kind === ActionType.Observe)
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/StepsView.scala]: Why are we reverting this? If we do so, let's remove the `FIXME`

Comment (jluhrs) [reply modules/seqexec/server/src/main/scala/seqexec/server/StepsView.scala]: This change does not address the `FIXME`. It is just a terser version of the same code.

Comment (cquiroz) [reply modules/seqexec/server/src/main/scala/seqexec/server/StepsView.scala]: Your right

**Pull request merged**